### PR TITLE
KalInput - Add custom icon

### PR DESCRIPTION
**Pull request by jsp-up**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.html [hunk 1]
```diff
@@ -12,6 +12,11 @@
     clear
   </kal-icon>
 
+  <kal-icon *ngIf="icon && !clearable"
+            (click)="customIconClicked()">
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.html:16]: iconClicked

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.spec.ts [hunk 2]
```diff
@@ -165,5 +172,28 @@ describe('KalInputComponent', () => {
     expect(fixture.debugElement.query(By.css('.counter')).nativeElement.textContent.trim()).toEqual(`${text.length * 2} / ${max}`);
   });
 
+  it('should display a custom icon', () => {
+    const spy = spyOn(kalInputInstance, 'customIconClicked');
+
+    const icons = fixture.debugElement.queryAll(By.directive(KalIconComponent));
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.spec.ts:178]: should test : no icon displayed when icon not setted

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.ts [hunk 3]
```diff
@@ -41,8 +50,17 @@ export class KalInputComponent extends FormElementComponent<string> implements O
...
+  /**
+   * Custom icon to use for the input
+   */
+  @Input() icon: string;
+
   control: FormControl;
 
+  private controlChangedSubscription = Subscription.EMPTY;
+
+  private iconClicked = new EventEmitter();
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.ts]: should use @Output

**Review by ydomenjoud — CHANGES_REQUESTED** [review]

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.html (hunk 1 above)
Comment (jsp-up) [reply projects/kalidea/kaligraphi/src/lib/atoms/kal-input/kal-input.component.html:16]: I can"t rename because it's the name of my event emitter

**Review by MichaelGrosio — APPROVED** [review]

**Review by ydomenjoud — APPROVED** [review]

**Pull request merged**